### PR TITLE
Add deprecation note about X.509/SHA-1

### PR DESCRIPTION
**Pull request by swenson**
In prepration for moving to Go 1.18 in Vault 1.12.

### CODE REVIEW

File: website/content/docs/deprecation/faq.mdx [hunk 1]
```diff
@@ -35,3 +35,17 @@ These features were deprecated in prior releases of Vault. We are targeting the
...
+
+Starting with Vault 1.12.0, Vault will be built with Go 1.18.
+The Go 1.18 standard library [rejects X.509 certificates](https://tip.golang.org/doc/go1.18#sha1) whose signatures use a SHA-1 hash.
+
+If this issue impacts your usage of Vault, you can temporarily work around it by deploying Vault with the environment variable `GODEBUG=x509sha1=1` set.
+This workaround will fail in a future version of Go, but the Go team has not said when.
+
+Here are the use cases we that might use certificates with SHA-1 still:
+
+- AWS Credential Plugin: AWS [still supports SHA-1 signatures](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html).
```
[the remark this answers is not in the document]
Comment (swenson) [reply website/content/docs/deprecation/faq.mdx]: I updated to clarify.

**Review by sgmiller — APPROVED** [review]

File: website/content/docs/deprecation/faq.mdx [hunk 2]
```diff
@@ -35,3 +35,17 @@ These features were deprecated in prior releases of Vault. We are targeting the
...
 | AppID                 | [AppRole auth method](/docs/auth/approle)                                                                           |
 | Standalone DB engines | [Combined DB engines](/docs/secrets/databases)                                                                      |
+
+### Q: What is the impact of removing support for X.509 certificates with signatures that use SHA-1?
+
+Starting with Vault 1.12.0, Vault will be built with Go 1.18.
+The Go 1.18 standard library [rejects X.509 certificates](https://tip.golang.org/doc/go1.18#sha1) whose signatures use a SHA-1 hash.
+
+If this issue impacts your usage of Vault, you can temporarily work around it by deploying Vault with the environment variable `GODEBUG=x509sha1=1` set.
+This workaround will fail in a future version of Go, but the Go team has not said when.
```
Comment (taoism4504) [inline website/content/docs/deprecation/faq.mdx:45]:
```suggestion
This workaround will fail in a future version of Go, however, the Go team has yet to announce the Go version that will no longer require this workaround.
```

Comment (taoism4504) [reply website/content/docs/deprecation/faq.mdx:45]: Something along those lines?

File: website/content/docs/deprecation/faq.mdx [hunk 3]
```diff
@@ -35,3 +35,17 @@ These features were deprecated in prior releases of Vault. We are targeting the
...
+
+Here are the use cases we that might use certificates with SHA-1 still:
```
Comment (taoism4504) [inline website/content/docs/deprecation/faq.mdx:47]:
```suggestion
Here are the use cases that may still use certificates with SHA-1:
```

**Review by taoism4504 — COMMENTED** [review]
Just a few suggestions.

File: website/content/docs/deprecation/faq.mdx (hunk 2 above)
Comment (cipherboy) [reply website/content/docs/deprecation/faq.mdx:45]: That's not strictly correct; at the point in time when this workaround is removed, the verification will fail entirely and the only way to work will be on an older Vault version. 

Maybe this is better?

```suggestion
This workaround will fail in a future version of Go, however, the Go team has not said when they will remove this workaround.
```

**Pull request merged**